prisma/schema.prisma

```diff
@@ -19,8 +19,45 @@ model User {
   deliveryLogs DeliveryLog[]
   posts        Post[]
   profile      Profile?
+  grades       Grade[]
 }
 
+model Exercise {
+  id          Int        @id @default(autoincrement())
+  name        String
+  description String?
+  unitId      Int
+  unit        Unit       @relation(fields: [unitId], references: [id])
+  grades      Grade[]
+}
+
+model Unit {
+  id          Int        @id @default(autoincrement())
+  name        String
+  description String?
+  moduleId    Int
+  module      Module     @relation(fields: [moduleId], references: [id])
+  exercises   Exercise[]
+}
+
+model Module {
+  id          Int        @id @default(autoincrement())
+  name        String
+  description String?
+  units       Unit[]
+}
+
+model Grade {
+  id           Int       @id @default(autoincrement())
+  exerciseId   Int
+  exercise     Exercise  @relation(fields: [exerciseId], references: [id])
+  userId       Int
+  user         User      @relation(fields: [userId], references: [id])
+  grade        Int
+  completedAt  DateTime  @default(now())
```

Comment:
https://www.prisma.io/docs/orm/prisma-schema/data-model/models#defining-a-unique-field

This table Grade might need a `@@unique` constraint for the exerciseId and userId so that 1 user can complete 1 exercise only once on the table and not appear twice on this table:

```js
// include this inside Grade model
// Ensure unique combination of user and exercise
@@unique([exerciseId, userId])
```